### PR TITLE
[Refactor:Calendar] Luxon Date Objects

### PR DESCRIPTION
**Pull request by oskurtt**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

We use the builtin JS date library in a number of places. While this mostly works, we keep getting bit by browser implementation specific stuff and pitfalls because it's not a very developer friendly interface. This is especially a bit hairy as we then do bits of math with these dates, which is equally error prone. Many of the strings passed in do not follow the ISO8601 format and rely on non-standard browser behavior to parse the date.

### What is the new behavior?
Instead of using the builtin JS date library, we will be using Luxon. I have only made this change in latedays.js so the team can review and approve of integrating Luxon for the rest of Submitty.

### CODE REVIEW

**Review by ziesski — CHANGES_REQUESTED** [review]
Please follow our title guideline, which can be found [here](https://submitty.org/developer/getting_started/make_a_pull_request)

**Comment by williamjallen** [comment]
@oskurtt Your commit messages are not very informative, and it's hard to tell why a given change was made.  Single-letter commit messages, and commit messages like "update foo.txt" should be avoided.  Please refer to the slides from my presentation a few weeks ago for more information.

**Review by williamjallen — COMMENTED** [review]
Luxon seems like a modern library, with a healthy development process and release cycle.  I support adding Luxon for client-side date handling.

File: site/public/js/latedays.js
[the remark this answers is not in the document]
Comment (oskurtt) [reply site/public/js/latedays.js:1]: I will create another branch with only the dependency installed

[the remark this answers is not in the document]
Comment (williamjallen) [reply site/public/js/latedays.js:1]: @oskurtt It looks like you created https://github.com/Submitty/Submitty/pull/9890 to resolve this comment.  Please make sure you link any relevant PRs and issues like this so it's easier to track down things like this in the future.

[the remark this answers is not in the document]
Comment (oskurtt) [reply site/public/js/latedays.js:1]: If given the green light, I can continue working on refactoring files that needs to use Luxon. I will request another review for this PR when I have that done. In the meantime I would appreciate if someone can merge https://github.com/Submitty/Submitty/pull/9890 in the meantime.

**Pull request closed without merging**